eox_nelp/utils.py

```diff
@@ -78,3 +83,39 @@ def is_valid_national_id(national_id, raise_exception=False):
         )
 
     return check_national_id
+
+
+def extract_course_id_from_string(string):
+    """This return a sub-string that matches the course_ir regex
+
+    Arguments:
+        string: This is a string that could contains a sub-string that matches with the course_id form.
+
+    Returns:
+        course_id <string>: Returns course id or an empty string.
+    """
+    matches = re.search(COURSE_ID_REGEX, string)
+
+    if matches:
+        return matches.group()
+
+    return ""
+
+
+def get_course_from_id(course_id):
+    """
+    Get Course object using the `course_id`.
+
+    Arguments:
+        course_id (str) :   ID of the course
+
+    Returns:
+        Course
+    """
+    course_key = CourseKey.from_string(course_id)
+    course_overviews = CourseOverview.get_from_ids([course_key])
```

Comment:
Why if the methods is called `get_course_from_id` (from one id), you use a method for multiple ids instead of the method for [one](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/content/course_overviews/models.py#L388)??

Reply:
This method is a copy of [this](https://github.com/openedx/event-routing-backends/blob/master/event_routing_backends/helpers.py#L138) which implements  this line `course_overviews = get_course_overviews([course_key])` however I made a mistake  and I copied the logic from the method [get_course_overviews_from_ids](https://github.com/eduNEXT/edunext-platform/blob/ednx-release/mango.master.nelp/openedx/core/djangoapps/content/course_overviews/api.py#L61C5-L61C34) instead of the [get_course_overviews](https://github.com/eduNEXT/edunext-platform/blob/ednx-release/mango.master.nelp/openedx/core/djangoapps/content/course_overviews/api.py#L74C5-L74C25) method

Reply:
This method is a copy of [this](https://github.com/openedx/event-routing-backends/blob/master/event_routing_backends/helpers.py#L138) which implements  this line `course_overviews = get_course_overviews([course_key])` however I made a mistake  and I copied the logic from the method [get_course_overviews_from_ids](https://github.com/eduNEXT/edunext-platform/blob/ednx-release/mango.master.nelp/openedx/core/djangoapps/content/course_overviews/api.py#L61C5-L61C34) instead of the [get_course_overviews](https://github.com/eduNEXT/edunext-platform/blob/ednx-release/mango.master.nelp/openedx/core/djangoapps/content/course_overviews/api.py#L74C5-L74C25) method

Reply:
I copied the logic because we already have the CourseOverview model and was easier  however the other logic requires a backend, so it will be easier to implement new api backend